src/I18nExtension.php

```diff
@@ -5,42 +5,66 @@
  *
  * (c) 2010-2019 Fabien Potencier
  * (c) 2019 phpMyAdmin contributors
+ * (c) 2021 Christopher Marshall contributor
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
  */
 
-namespace PhpMyAdmin\Twig\Extensions;
+namespace PhpMyAdmin\Twig\Extensions
+{
+	use PhpMyAdmin\Twig\Extensions\TokenParser\TransTokenParser;
+	use Twig\Extension\AbstractExtension;
+	use Twig\TwigFilter;
 
-use PhpMyAdmin\Twig\Extensions\TokenParser\TransTokenParser;
-use Twig\Extension\AbstractExtension;
-use Twig\TwigFilter;
+	class I18nExtension extends AbstractExtension
+	{
+		/**
+		 * {@inheritdoc}
+		 */
+		public function getTokenParsers()
+		{
+			return [new TransTokenParser()];
+		}
 
-class I18nExtension extends AbstractExtension
-{
-    /**
-     * {@inheritdoc}
-     */
-    public function getTokenParsers()
-    {
-        return [new TransTokenParser()];
-    }
+		/**
+		 * {@inheritdoc}
+		 */
+		public function getFilters()
+		{
+			return [
+				new TwigFilter('trans', '\twig_gettext_translate'), /* Note, the filter does not handle plurals */
+			];
+		}
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getFilters()
-    {
-        return [
-            new TwigFilter('trans', 'gettext'),
-        ];
-    }
+		/**
+		 * {@inheritdoc}
+		 */
+		public function getName()
+		{
+			return 'i18n';
+		}
+	}
+}
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getName()
-    {
-        return 'i18n';
-    }
+namespace
+{
+	/**
+	 * Translate a GetText string via filter
+	 *
+	 * @param string| $message The message to translate
+	 * @param string|null $domain The GetText domain
+	 *
+	 * @return string
+	 */
+	function twig_gettext_translate($message, $domain = null)
```

Comment:
Only this file diff and this function and it's call look out of the project scope
I am not sure that calling gettext functions is something that should be done here
Can you explain ?

Reply:
Sure, so that is for the trans filter. Originally, the filter contained only a call to the gettext function when it was instantiated in this file. Because it was only a simple function call, there was no need for a new function at that time. When I changed it to allow calling the domain, it needed to be done in a new function to check whether there was a domain or not.

I recognise the concerns of it not seeming right, as I initially felt that too but when I looked deeper into it I felt the performance gain was so marginal by including that in a new file that it simply wasn't worth it. I also modelled it off existing Twig exensions, such as the **IntlExtension** here: https://github.com/twigphp/Twig-extensions/blob/master/src/IntlExtension.php

That extension uses exactly the same layout as what I've done here, along with a couple of the other ones in the old repisitory this extension was part of.